agent/agent.go

```diff
@@ -138,29 +138,7 @@ func (a *GossipAgent) Listen(ipString string, port int) error {
 	a.Logger.Info("agent successfully subscribed", zap.String("topic", topicName))
 
 	// read messages and print stats to logger
-	go func() {
-		for {
-			raw, err := sub.Next(context.Background())
-			if err != nil {
-				fmt.Printf("Peer %v error receiving message on topic: %v\n", host.ID(), err)
-				continue
-			}
-
-			// read message data from received bytes
-			var data packet
-			if err := readMessage(raw.Data, &data); err != nil {
-				fmt.Printf("Peer %v error reading message ID: %v\n", host.ID(), err)
-
-			}
-			a.Logger.Info("message received",
-				zap.String("peer", host.ID().Pretty()),
-				zap.String("direction", "received"),
-				zap.String("msgID", data.MessageID.String()),
-				zap.Bool("aggregateStats", data.IsUseful),
-				zap.String("from", raw.ReceivedFrom.Pretty()),
-			)
-		}
-	}()
+	go a.readMessagesLoop(sub)
```

Comment:
Please don't use this.
Leave the inline goroutine, or create a function that doesn't belong to a specific type